logstash-core/src/main/java/org/logstash/execution/ConvergeResultExt.java

```diff
@@ -125,6 +125,8 @@ public static ActionResultExt create(final ThreadContext context, final IRubyObj
                 result = new SuccessfulActionExt(context.runtime, RubyUtil.SUCCESSFUL_ACTION_CLASS).initialize(context);
             } else if (actionResult.eql(context.fals)) {
                 result = FailedActionExt.fromAction(context, RubyUtil.FAILED_ACTION_CLASS, action, actionResult);
+            } else if (actionResult.getMetaClass().getName().equals("Java::JavaLang::IllegalStateException")) {
```

Comment:
Note for reviewers:
without this addition we can't catch compilation exception generated from https://github.com/elastic/logstash/blob/de5888ba18038156cd76bac306358d9a351295ce/logstash-core/src/main/java/org/logstash/config/ir/CompiledPipeline.java#L119

Reply:
believe this is smt worth looking into and discussing whether we have other options, 
I do not want to grow the scope here just that it looks a bit smelly that we need this ...

Reply:
I agree with you, is not clean ans smell, but I want's able to grab the original exception class from `actionReult.getMetaClass()` so that I could do something like:
``java
} else if (actionResult.getMetaClass() instanceof IllegalStateException) {
...
}
```
that's far from perfection, but at least is not like a magic constant. Perhaps we should create a `ConfigPipelineCompilationException` instead of throw a more general `IllegalStateException`

Reply:
this condition was a result of trying to resolve some of the String to Symbol conversions in tests and code